Library/Formula/gdb.rb

```diff
@@ -25,12 +25,13 @@ class Gdb < Formula
   end
 
   option "with-brewed-python", "Use the Homebrew version of Python"
+  option "with-brewed-readline", "Use the Homebrew version of Readline"
   option "with-version-suffix", "Add a version suffix to program"
   option "with-all-targets", "Build with support for all targets"
 
   depends_on "pkg-config" => :build
-  depends_on "readline"
-  depends_on "xz"
+  depends_on "python" if build.with? "brewed-python"
+  depends_on "readline" if build.with? "brewed-readline"
```

Comment:
In that case you can/should fill out the option description to explain what it's doing.